cdist/config.py

```diff
@@ -176,8 +177,15 @@ def commandline(cls, args):
                               " ".join(failed_hosts))
 
     @classmethod
-    def _resolve_remote_cmds(cls, args, host_base_path):
-        control_path = os.path.join(host_base_path, "ssh-control-path")
+    def _resolve_ssh_control_path(cls):
+        base_path = tempfile.mkdtemp()
+        control_path = os.path.join(base_path, "s")
```

Comment:
@telmich This is called when constructing remote_exec/copy commands. And this is done only once for a host. remote_exec/copy commands are constructed from the pattern which contains placeholder for control path (if mux options inspection detected that options are supported).
So control path is 'stored' in remote_exec/copy constructed commands.